app/models/Service.scala

```diff
@@ -64,9 +96,216 @@ object Service {
+      val uploadFuture = uploadEnvironment(dataCsv).flatMap {
+        uploaded => uploadService(dataCsv)
+      }
+      Right(Await.result(uploadFuture, 10.seconds))
+    } else {
+      Left(ErrorUploadCsv(s"First column ${dataCsv(0)} is not recognized as ${csvKey} "))
+    }
+
+  }
+
+  /**
+    * Upload a new Environment from the service
+    *
+    * @param dataCsv
+    */
+  private def uploadEnvironment(dataCsv: Array[String]): Future[Boolean] = {
+
+    val envS = dataCsv(csvTitle.indexOf("environmentName")).trim
+    val localTarget = dataCsv(csvTitle.indexOf("localTarget"))
+    if (envS.isEmpty) {
+      throw new Exception(s"environmentName is mandatory when uploading the service ${localTarget}.")
+    }
+
+    def insertEnvironment(env: Environment, localTarget: String): Future[Boolean] = {
+      val u = Environment.insert(env)
+      u.map {
+        case res => {
+          Logger.info(s"Created new default environment ${env.name} for service ${localTarget}")
+          true
+        }
+      }
+    }
+
+    // Search the environment by its name
+    val potentialEnvironmentF = Environment.findByName(envS, cached = false)
+    potentialEnvironmentF.flatMap {
+      case Some(e) => {
+        // Environment exists, so we do nothing because we don't have any other info to update
+        Logger.info(s"Environment ${e.name} exists for service ${localTarget}")
+        Future.successful(true)
+      }
+      case None => {
+        // Create a default environment with name
+        Logger.info(s"Environment ${envS} does not exist for service ${localTarget}")
+        val env = new Environment(
+          None,
+          envS,
+          List()
+        )
+        insertEnvironment(env, localTarget)
+      }
+    }
+  }
+
+
+  /**
+    * Check if service already exist (with localTarget and Environment). Insert or update if exist.
+    *
+    * @param dataCsv line in csv file
+    * @return service (new or not)
+    */
+  private def uploadService(dataCsv: Array[String]): Future[Boolean] = {
+
+    val id = dataCsv(csvTitle.indexOf("id"))
+    val typeRequest = dataCsv(csvTitle.indexOf("typeRequest"))
+    val localTarget = dataCsv(csvTitle.indexOf("localTarget"))
+    val environmentName = dataCsv(csvTitle.indexOf("environmentName"))
+    val httpMethod = dataCsv(csvTitle.indexOf("httpMethod"))
+    val potentialServiceF = if (id.isEmpty) {
+      findByLocalTargetAndEnvironmentName(typeRequest, localTarget, environmentName, HttpMethod.valueOf(httpMethod))
+    } else {
+      findById(dataCsv(csvTitle.indexOf("environmentName")), id)
+    }
+
+    def insertService: Future[Boolean] = {
+      // Insert the service by generating the new id
+      val service = getServiceFromCSV(dataCsv).copy(_id = Some(BSONObjectID.generate))
+      val insert = Service.insert(service)
+      insert.map {
+        case res => Logger.info(s"Created new service ${localTarget} for ${environmentName} and ${typeRequest}/${httpMethod}")
+          true
```

Comment:
Whenever you're using `Future[Boolean]` but always return `true` if the future succeeds, you should instead use `Future[Unit]`